### ISSUE TITLE
Task exists error with Dependency-Submission action

### ISSUE THREAD

**Issue by nnellanspdl** [gradle/actions#73]
Hello, I am trying to use the following steps in my GitHub Workflow.  This is all I'm doing, yet I'm getting an error:

```yaml
- uses: actions/checkout@v4
- name: Submit Dependencies
  env:
    GITHUB_JOB_CORRELATOR: '${{ github.workflow }}-${{ github.job }}'
    GITHUB_JOB_ID: '${{ github.run_id }}'
  uses: gradle/actions/dependency-submission@v3
  with:
    dependency-graph: 'generate-and-submit'
```

When running this I get the following error:
```
* What went wrong:
Cannot add task 'ForceDependencyResolutionPlugin_resolveAllDependencies' as a task with that name already exists.
```

**Comment by bigdaz** [comment]
This is odd. Can you share any more context about your project?

Are you by any chance applying the [Dependency Graph Plugin](https://github.com/gradle/github-dependency-graph-gradle-plugin) directly to your project?
Is this in a separate workflow Job, or is it combined with other workflow steps?
Is there a reason you're not simply following [the instructions here](https://github.com/gradle/actions/blob/main/dependency-submission/README.md)?

Can you please try the plugin in isolation, by following these instructions to generate a standalone dependency report?

**Comment by bigdaz** [comment]
Also, please provide the full exception by adding `--stacktrace` as an additional argument. 
https://github.com/gradle/actions/blob/main/dependency-submission/action.yml#L32-L34

**Comment by nnellanspdl** [comment]
Here's my entire workflow:

```yaml
name: Gradle Build

on:
  push:
    branches: [ feature/**, bugfix/**, hotfix/** ]
  pull_request:
    branches: [ main, release/** ]

jobs:

  dependency:
    runs-on: self-hosted
    steps:
      - uses: actions/checkout@v4
      - name: Submit Dependencies
        env:
          GITHUB_JOB_CORRELATOR: '${{ github.workflow }}-${{ github.job }}'
          GITHUB_JOB_ID: '${{ github.run_id }}'
        uses: gradle/actions/dependency-submission@v3
        with:
          additional-arguments: '--stracktrace'
```

I had to manually set those 2 variables.  When I run without them I get errors that they don't exist and that I need to set them.

I have a hunch something is up with our github runners.  We are running our own self-hosted, long-lived runner VMs.  Meaning, they are used over and over for many builds without being torn down & rebuilt each time.  Could that somehow be causing me problems?

And here's the stacktrace output
```
Run gradle/actions/dependency-submission@v3
Run if [ -n "${GRADLE_BUILD_ACTION_SETUP_COMPLETED}" ]; then
Run gradle/actions/setup-gradle@v3.1.0
Gradle User Home already exists: will not restore from cache.
Merged default JDK locations into /home/opsadm/.m2/toolchains.xml
Enabling dependency graph generation
/home/opsadm/actions-runner/_work/xxx/xxx/gradlew -Dorg.gradle.configureondemand=false -Dorg.gradle.dependency.verification=off -Dorg.gradle.unsafe.isolated-projects=false :ForceDependencyResolutionPlugin_resolveAllDependencies --stracktrace
Starting a Gradle Daemon, 3 incompatible and 2 stopped Daemons could not be reused, use --status for details
Generating dependency graph into '/home/opsadm/actions-runner/_work/xxx/xxx/dependency-graph-reports/Gradle Build-dependency.json'
Generating dependency graph into '/home/opsadm/actions-runner/_work/xxx/xxx/dependency-graph-reports/gradle_build-dependency.json'
Generating dependency graph into '/home/opsadm/actions-runner/_work/xxx/xxx/dependency-graph-reports/gradle_build-dependency.json'

GitHubDependencyGraphPlugin: Wrote dependency snapshot to 
/home/opsadm/actions-runner/_work/xxx/xxx/dependency-graph-reports/Gradle Build-dependency.json
Gradle Build did not complete successfully: Dependency Graph file will not be generated.
Constructing renderer: org.gradle.github.dependencygraph.GitHubDependencyGraphRenderer

GitHubDependencyGraphRenderer: Wrote dependency snapshot to 
/home/opsadm/actions-runner/_work/xxx/xxx/dependency-graph-reports/gradle_build-dependency.json

FAILURE: Build failed with an exception.

* What went wrong:
Cannot add task 'ForceDependencyResolutionPlugin_resolveAllDependencies' as a task with that name already exists.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.4/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 7s
Error: Gradle build failed: see console output for details
```

**Comment by nnellanspdl** [comment]
I have it working better now ... I think.

I added a step to first clear the gradle user home directory `~/.gradle`, in order to clear all the crud from all the different build jobs these runners do.  If I were using GitHub-hosted runners I wouldn't have to worry about this.

There was also a GitHub Actions Caching step, which reading the docs, may be a bad idea in combination with the dependency-submission action.  So, I removed the extra cache step.

After both of those changes it's working much better, and I no longer have to manually specify the environment variables on the step, either.
